extension/background/background.js

```diff
@@ -11,6 +13,73 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
     }
 })
 
+function setUpWebRequestOriginRemoval() {
+    chrome.permissions.contains({ 
+        permissions: ['webRequestBlocking'],
+        origins: ['https://www.youtube.com/']
+    }, (permissionExists) => {
+        if (permissionExists) { // In Firefox (Manifest V2)
+            // YouTube declines requests with wrong Origin.
+            // We have to remove the Origin header which is added automatically by the browser.
+            chrome.webRequest.onBeforeSendHeaders.addListener(
+                details => {
+                    const newRequestHeaders = details.requestHeaders.filter(header => {
+                        return header.name.toLowerCase() !== "origin"
+                    })
+                    return {requestHeaders: newRequestHeaders}
+                },
+                {urls: ["https://www.youtube.com/*"]},
+                ["blocking", "requestHeaders", chrome.webRequest.OnBeforeSendHeadersOptions.EXTRA_HEADERS].filter(Boolean)
+            )
+        } // In Chrome/Chromium (Manifest V3), Origin is removed via declarative net request
+    })    
+}
+
 async function fetchChapters(videoId) {
     return await youtubei.fetchChapters(videoId)
 }
+
+// time-comments related code below
+
+export async function fetchChaptersFromComments(videoResponse) {
+    const comments = await youtubei.fetchComments(videoResponse)
+
+    // Currently using only the first minimally suitable comment.
+    // Maybe later implement more sophisticated comment selection.
+    for (let i = 0; i < comments.length; i++) {
+        if (!comments[i].isPinned) {
+            continue
+        }
+        const tsContexts = getTimestampContexts(comments[i].text)
+        if (tsContexts.length) {
```

Comment:
There should be at least 2 chapters IMO.

Reply:
Done.